### PR TITLE
Add local fixtures in test_polydata

### PR DESCRIPTION
**Pull request by adeak**
The `test_polydata.py` module had a handful of global meshes that were being copied all over the place. This PR replaces them with module-level fixtures.

One non-trivial change is that the now-local `sphere` fixture (coming from `pv.Sphere`) shadows the global `sphere` fixture (coming from `pv.examples.load_sphere`, ultimately reading a file with data arrays). This doesn't seem like an issue and the tests run fine.

### CODE REVIEW

**Comment by adeak** [comment]
GitHub actions are [still flaky](https://www.githubstatus.com/). We can wait until they come back; this one needs solid CI.

**Review by akaszynski — APPROVED** [review]
I restarted CI and it seems that it ran fine.  Test run locally with `pytest -vx -n 4` on Linux.

I see no issue using a different sphere fixture other than we might consider replacing the global fixture in a later PR with one like this (if we're fine giving up the data_arrays).

**Comment by akaszynski** [comment]
You've got merge permission and can fire when ready.

**Comment by adeak** [comment]
Thanks!

**Pull request merged**
